### ISSUE TITLE
DEBUG logging calls are stored in release binary/Change logging system

### ISSUE THREAD

**Issue by username223** [OpenVR-Advanced-Settings/OpenVR-AdvancedSettings#389]
**Describe the bug**
DEBUG logging calls are compiled into the final binary, increasing binary size.

**To Reproduce**
1. Compile without `debugSymbolsAndLogs`.
2. Run `strings AdvancedSettings | grep Pulse` on the final binary.
3. Notice that DEBUG messages are still there.

**Expected behavior**
DEBUG calls would not be in the final binary.

**Additional context**
It might be worth looking into redoing the logging system.

Currently we are directly using the [easylogging++](https://github.com/amrayn/easyloggingpp) library. It is currently looking for maintainers.

It has the advantages of being relatively simple to use, appending text with `operator<<` and generally being able to automatically convert objects to text that makes sense.

It has the disadvantage of not being able to ifdef out DEBUG statements at compile time, or use Python like formatting in strings.

Changing the loggging system to use something like [fmt](https://github.com/fmtlib/fmt) for the frontend and remaining on easylogging++ for the backend would give the benefits of being able to do python/printf formatting in logging messages, while still being able to quickly change logging libraries or compile out certain types of logging.

`fmt` is also being added to `std` in C++20, so if we upgrade to C++20 we will have it by default.

**Comment by ykeara** [comment]
Makes sense, and would produce more human read-able logs, which could help us.

so

Pros:
- better formatting/human readable logs
- reduced executable size? (better perf?)

Cons:
-  are we going to have issues on windows with `fmt` and microsoft redistributable c++ pkgs? especially if we move forward to c++ 20 immediately?

Rationale:
- are we gaining performance are executable size is small enough as is that the extra strings don't really matter imo, but if we do take performance loss because the calls are still compiled to code then I  support the change in the long term, the problem being the possibility of library issues in the short term especially with expected platforms.

**Comment by username223** [comment]
@ykeara 
C++20 is not ready on any compiler set yet, so we won't be switching before at least June or July.
There shouldn't be any problems with C++ redists, at least not that I'm aware.

A pro would also be that we could move away from easylogging++ if it became deprecated.
